federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -154,7 +170,9 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 		deleted := cc.knownClusterSet.Difference(observedSet)
 		for clusterName := range deleted {
 			glog.V(1).Infof("ClusterController observed a Cluster deletion: %v", clusterName)
+			// delete the cluster name from knownClusterSet and delete the cluster client from clusterKubeClientMap
```

Comment:
same. how about calling delFromClusterSet here? You can also move glog.V(1) observed a cluster deletion to that function